### ISSUE TITLE
Server-side memory leak

### ISSUE THREAD

**Issue by TheInfamousAlk** [AllTheMods/ATM-6#1007]
Version 1.5.1. (also happened with 1.5.0) Server hosted via akliz.net, with 8gb ram.
**Describe the bug**
About every 5ish hours, the server I'm hosting via akliz.net crashes due to running out of memory. 

**To Reproduce**
No steps, just keep the server on for several hours and it crashes. 

**Additional context**
First off, I believe the game itself isn't crashing, but it is running out of memory has has to be killed. The java memory limit is 8, but the server reaches that, and has to be killed. This means I cannot find a crash log, only debug and regular logs.

Version 1.5.1. (also happened with 1.5.0) Server hosted via akliz.net, with 8gb ram. Stock settings except 3: "world-type=quark\:realistic", "allow-flight=true" and in world\serverconfig\minecolonies-server.toml I changed allowinfinitesupplychests = true . 

The akliz support said they have not seen this specific issue out of hundreds of ATM6 servers they are running, so my best bet is to submit a bug report here.

**Logs**
Here is the github paste link. I took files from today and yesterday. The latest crash was today. Let me know if I should provide more, older logs. I have them going back like 3 days and 3 or 4 crashes. 

https://gist.github.com/TheInfamousAlk/ab552243eadf42aadb4608955d7b9ad3

**Comment by TheInfamousAlk** [comment]
Update: the akliz support staff said they have 4 other servers doing the same thing, and all have the quark:realistic level type enabled. It's likely related to that. But it doesn't happen on single player, and I just tested a server with only quark on it (same version) locally and I did not have the memory leak. So either it's another mod or setting or something in ATM6 causing the leak, or it's something to do with akliz's configuration, using a newer java or something. (They said they use java 11 for their ATM6 servers.)

**Comment by TheInfamousAlk** [comment]
Another update: I wiped the world, started over with biomesoplenty level type, and the memory leak showed up again immediately.

**Comment by whatthedrunk** [comment]
Have you tried running with default settings? We currently use those for all our test worlds and public servers as we had issues with BOP in the past but dont want to remove it as it would break too many worlds.

**Comment by TheInfamousAlk** [comment]
Oh, the akliz support said several people were using BOP without issue. I have not tried default, but both BOP and realistic had the same memory leak so I'm not sure if it's being caused by something else, or by both of those world types independently. I can try to make a default world to test with today, see if it happens still.

**Comment by TheInfamousAlk** [comment]
I tried with completely default settings, and it has happened again. The only thing I can think of is we have used minecolonies supplycamps as initial bases every time. While this doesn't spawn NPCs or create a colony, it's the only thing we've done each time. I'm going to try again and ban all use of minecolonies items to see if it makes a difference.

**Comment by TheInfamousAlk** [comment]
I tried again and banned minecolonies' items, still the memory leak. Completely default config. I am giving up the testing and just putting up the BOP world we started on, and maybe set it to restart 3 or 4 times a day (or more if I have to). Let me know if there's any specific testing I should do that doesn't involve making a brand new server or wiping our world.

**Comment by ScoutsGoneWild** [comment]
I'm experiencing the same issue on my ScalaCube server, this is not a host problem it would seem. Below is a graph of my server's ram use, I installed the pack on February 6th, and started it at 2:40 pm CST today which is the start of the first climb onto the right side of the graph, the server crashed and started to climb again before crashing at 9:40 pm. The server is capped at 7.2GB of ram, but can run over into 8 if needed, which it did and crashed upon reaching 8GB. The server is also experiencing severe lag but is not tps (checked with /forge tps) or connection-related (have run the server before at 3mb/s this was far less traffic than that showed by a network traffic graph.) The only change I have made to the server is setting realistic world gen. I suspect a memory leak, but I am no expert at minecraft/ modpack development or ram.

![](https://cdn.discordapp.com/attachments/807848498280005663/808177935071969300/unknown.png)
